src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/FileUriProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    internal abstract class FileUriProvider
+    {
+        public abstract Uri GetOrCreate(ITextBuffer textBuffer);
```

Comment:
>GetOrCreate [](start = 28, length = 11)

Nit: again, a comment would be nice and what sort of Uri this provides (I know we talked about it yesterday, still might be nice to have in the code)